### PR TITLE
Support truecolor and background transparency

### LINKED ISSUE #10: True color support and background transparency support
Hi,

If 'set termguicolors' is set in .vimrc, the colors will not be displayed normally.

In additon, could you consider supporting background transparency feature? 

Thanks~

### PR DESCRIPTION
**Pull request by windvalley**
Fixes #10

Option: `let g:colorscheme_primary_bg_transparency = 1`

Before:
dark background

After:
dark background transparency

### CODE REVIEW

**Comment by windvalley** [comment]
@googlebot I signed it!

File: colors/primary.vim [hunk 1]
```diff
@@ -25,8 +25,9 @@ if exists('syntax_on')
 endif
 let g:colors_name = 'primary'
 let s:disable_italic = get(g:,'colorscheme_primary_disable_italic', 0)
+let s:enable_bg_transparency = get(g:,'colorscheme_primary_bg_transparency', 0)
```
Comment (dbarnett) [inline colors/primary.vim:28]: For the variable name here, I'm wondering, have you seen any other colorschemes do similar, and do they have any existing variable name we could adopt as a convention? If not, this is fine as-is.

**Comment by dbarnett** [comment]
Changes look reasonable (and nice screenshot!). Could you detail for the PR how the specific changes work since colorscheme mechanics can be a little cryptic? It looks like you've pulled out the bg color to only be defined on `Normal` and `NonText` (while making them conditional), and then they get inherited by all the other syntax types?

@LisieRae Could you take a look as well?

File: colors/primary.vim (hunk 1 above)
Comment (windvalley) [reply colors/primary.vim:28]: Hi,

There seems to be no convention about this option, for example:

gruvbox: 
g:gruvbox_transparent_bg
https://github.com/morhetz/gruvbox/wiki/Configuration#ggruvbox_transparent_bg

solarized: 
g:solarized_termtrans
https://github.com/altercation/vim-colors-solarized

So, can we use this `g:colorscheme_primary_enable_transparent_bg` ?

**Comment by windvalley** [comment]
> Changes look reasonable (and nice screenshot!). Could you detail for the PR how the specific changes work since colorscheme mechanics can be a little cryptic? It looks like you've pulled out the bg color to only be defined on `Normal` and `NonText` (while making them conditional), and then they get inherited by all the other syntax types?
> 
> @LisieRae Could you take a look as well?

You're right. Reference content:

`:help hi-normal`
```text
If you are running in the GUI, you can get white text on a black background with:
     highlight Normal guibg=Black guifg=White
```
`:help hi-normal-cterm`

```text
	When setting the "ctermfg" or "ctermbg" colors for the Normal group,
	these will become the colors used for the non-highlighted text.
	Example: 
		:highlight Normal ctermfg=grey ctermbg=darkblue
	When setting the "ctermbg" color for the Normal group, the
	'background' option will be adjusted automatically, under the
	condition that the color is recognized and 'background' was not set
	explicitly.  This causes the highlight groups that depend on
	'background' to change!  This means you should set the colors for
	Normal first, before setting other colors.
```

File: colors/primary.vim (hunk 1 above)
Comment (dbarnett) [reply colors/primary.vim:28]: Yep, this works for me, and the cross-references are really helpful, thanks!

File: doc/colorscheme-primary.txt [hunk 2]
```diff
@@ -42,6 +42,9 @@ does not support 256 colors. No problem - just remove that line and consult
 If you want to disable italic font, add this line to your .vimrc: >
   let g:colorscheme_primary_disable_italic = 1
 <
+                                        *g:colorscheme_primary_bg_transparency*
+If you want to enable background transparency, add this line to your .vimrc: >
+  let g:colorscheme_primary_bg_transparency = 1
```
Comment (dbarnett) [inline doc/colorscheme-primary.txt:47]: Syntax nit: Isn't this missing the `<` to explicitly mark the end of the block?

File: colors/primary.vim [hunk 3]
```diff
@@ -93,49 +94,56 @@ endif
...
-exe 'hi Operator     '.s:M.'bg='.s:BG  .' '.s:M.'fg='.s:GREY1.' '.s:M.'=bold'
-exe 'hi Tag          '.s:M.'bg='.s:BG  .' '.s:M.'fg='.s:GREY1.' '.s:M.'=bold'
-exe 'hi Ignore       '.s:M.'bg='.s:BG  .' '.s:M.'fg='.s:GREY1.' '.s:M.'=none'
-exe 'hi Underlined   '.s:M.'bg='.s:BG  .' '.s:M.'fg='.s:GREY1 .' '.s:M.'=underline'
+if (s:enable_bg_transparency == 1)
+  exe 'hi Normal       '.s:M.'bg=NONE '.s:M.'fg='.s:BLUE .' '.s:M.'=none'
+  exe 'hi NonText      '.s:M.'bg=NONE '.s:M.'fg='.s:GREY1.' '.s:M.'=bold'
+else
+  exe 'hi Normal       '.s:M.'bg='.s:BG .' '.s:M.'fg='.s:BLUE .' '.s:M.'=none'
+  exe 'hi NonText      '.s:M.'bg='.s:BG   .' '.s:M.'fg='.s:GREY1.' '.s:M.'=bold'
```
Comment (LisieRae) [inline colors/primary.vim:102]: Thanks for contributing! Looks like there are some whitespace inconsistencies on lines 98-102. Please fix.

File: doc/colorscheme-primary.txt (hunk 2 above)
Comment (windvalley) [reply doc/colorscheme-primary.txt:47]: Yep, I'll fixed later, thanks!

File: colors/primary.vim (hunk 3 above)
Comment (windvalley) [reply colors/primary.vim]: Hi, I already fixed it.

File: doc/colorscheme-primary.txt (hunk 2 above)
Comment (windvalley) [reply doc/colorscheme-primary.txt]: > Syntax nit: Isn't this missing the `<` to explicitly mark the end of the block?

Hi, I already fixed it, and have changed the option to `g:colorscheme_primary_enable_transparent_bg`, please review the codes again, thanks!

File: doc/colorscheme-primary.txt [hunk 4]
```diff
@@ -42,6 +42,10 @@ does not support 256 colors. No problem - just remove that line and consult
 If you want to disable italic font, add this line to your .vimrc: >
   let g:colorscheme_primary_disable_italic = 1
 <
+                                        *g:colorscheme_primary_bg_transparency*
```
Comment (dbarnett) [inline doc/colorscheme-primary.txt:45]: Update this tag to match the updated variable name.

File: colors/primary.vim (hunk 3 above)
Comment (dbarnett) [reply colors/primary.vim]: I didn't see a change here. Strange spacing still on `'bg='.s:BG   .' '` here and farther down on the `hi Underlined` line.

Sorry for the back-and-forth. If it's a pain to update now, I might do a pass over the file later to make a few overall readability tweaks.

Comment (windvalley) [reply colors/primary.vim]: I've formatted the file globally to make it more readable, and thanks for your patience:)

> I didn't see a change here. Strange spacing still on `'bg='.s:BG .' '` here and farther down on the `hi Underlined` line.
> 
> Sorry for the back-and-forth. If it's a pain to update now, I might do a pass over the file later to make a few overall readability tweaks.

Comment (dbarnett) [reply colors/primary.vim]: Looks good to me!

**Comment by dbarnett** [comment]
Looks good! Merging…

Thanks for contributing!

**Pull request merged**